tools/dmake.cpp

```diff
@@ -46,11 +46,27 @@ static void getDeps(const std::string &filename, std::vector<std::string> &depfi
     // Is the dependency already included?
     if (std::find(depfiles.begin(), depfiles.end(), filename) != depfiles.end())
         return;
-
     std::ifstream f(filename.c_str());
-    if (! f.is_open()) {
-        if (filename.compare(0, 4, "cli/") == 0 || filename.compare(0, 5, "test/") == 0)
+    if (! f.is_open()) { 
+        if (filename.compare(0, 4, "lib/") == 0) {
```

Comment:
there are copy paste here. It was many years since I really looked at this code so maybe I am missing something ... but how about this:

    if (! f.is_open()) {
            getDeps("lib" + filename.substr(filename.find('/')), depfiles);
            getDeps("externals" + filename.substr(filename.find('/')), depfiles);
            getDeps("externals/simplecpp" + filename.substr(filename.find('/')), depfiles);
            getDeps("externals/tinyxml" + filename.substr(filename.find('/')), depfiles);
            return;
    }

Reply:
I didn't test that, but it looks like infinite recursion (if the file is in external, we will check lib again and again).

It's perhaps less intuitive,  but I think only the if (lib) case needs to check the externals, since every other code path will end up there anyway if there is an external. Similarly, the "test/" case only needs to reverse into "cli/", since the next call will recurse into "lib/" anyway.

Maybe the "tools/" case should recurse into "cli/" too, since I guess tools can use headers from "cli/" too (even though there's perhaps not done yet).